### PR TITLE
PLAT-133234: Add disabled and spotlightDisabled props to agate/MediaPlayer

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [x] Documentation was verified or is not changed
* [x] UI test was passed or is not needed
* [x] Screenshot test was verified or is not needed
* [ ] This is an API breaking change

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Refactored MediaPlayer to handle disabled Spotlight. Updated props.
Updated storybook and tests.

### Links
[//]: # (Related issues, references)
PLAT-133234

### Comments

### CODE REVIEW

File: MediaPlayer/MediaPlayer.js [hunk 1]
```diff
@@ -77,6 +82,14 @@ const MediaPlayerBase = kind({
 		 */
 		currentTime: PropTypes.number,
 
+		/**
+		 * Removes interactive capability from this component.
+		 *
+		 * @type {Boolean}
+		 * @public
+		 */
+		disabled: PropTypes.bool,
```
Comment (daniel-stoian-lgp) [inline MediaPlayer/MediaPlayer.js:91]: although this prop is added in Mediaplayer, The MediaSlider and MediaControls can still be changed via click/ 5-way. I think you need to pass this props to MediaSlider and all Buttons in MediaControls

File: tests/ui/specs/MediaPlayer/MediaPlayer-specs.js [hunk 2]
```diff
@@ -95,4 +95,38 @@ describe('MediaPlayer', function () {
...
+
+	describe('spotlightDisabled', function () {
+		const mediaPlayer = Page.components.mediaPlayerSpotlightDisabled;
+
+		it('should not have the slider knob focused', function () {
+			expect(mediaPlayer.slider.isFocused()).to.not.be.true();
+		});
+
+		describe('5-way', function () {
+
```
Comment (daniel-stoian-lgp) [inline tests/ui/specs/MediaPlayer/MediaPlayer-specs.js:107]: remove empty line

File: tests/ui/specs/MediaPlayer/MediaPlayer-specs.js [hunk 3]
```diff
@@ -95,4 +95,38 @@ describe('MediaPlayer', function () {
...
+				expect(mediaPlayer.slider.isFocused()).to.not.be.true();
+
+				Page.spotlightDown();
+
+				expect(mediaPlayer.playButton.isFocused()).to.not.be.true();
+			});
+		});
+
+		describe('using pointer', function () {
+
```
Comment (daniel-stoian-lgp) [inline tests/ui/specs/MediaPlayer/MediaPlayer-specs.js:118]: remove empty line

**Comment by daniel-stoian-lgp** [comment]
PR description needs to have the latest template. Can you update it?

**Review by daniel-stoian-lgp — APPROVED** [review]
LGTM

**Review by adrian-cocoara-lgp — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -41,7 +42,6 @@ The following is a curated list of changes in the Enact agate module, newest cha
 - `agate/Input` prop `invalidMessage` to customize the tooltip message when `invalid` prop is true
 - `agate/Input` prop `size` to change the size of both input and icons (default large)
 - `agate/Item` prop `centered`, `disabled`, `inline`, `marqueeOn`, and `size`
-- `agate/Keypad` prop `spotlightDisabled` to disable 5-way navigation
```
Comment (jeonghee27) [inline CHANGELOG.md:44]: Why did you remove changelog of previous version ?

File: MediaPlayer/MediaControls.js [hunk 5]
```diff
@@ -31,6 +31,14 @@ const MediaControls = kind({
 	name: 'MediaControls',
 
 	propTypes: /** @lends agate/MediaPlayer.MediaControls.prototype */ {
+		/**
+		 * Removes interactive capability from buttons inside this component.
```
Comment (jeonghee27) [inline MediaPlayer/MediaControls.js:35]: Hm, This discription was written in a slightly different from the other sentences. (" Removes interactive capability ")
I hope it is a sentence similar to the existing ones.

Let's just write "disables"

 * Disables MediaControl buttons and becomes non-interactive.
or
*  Disables the media buttons.

 Or please write it in your way

File: MediaPlayer/MediaControls.js [hunk 6]
```diff
@@ -202,7 +210,7 @@ const MediaControls = kind({
 
 	},
 
-	render: ({badge, menuIcon, nextTrackIcon, onNext, onPause, onPlay, onPrevious, onRepeat, onShuffle, paused, pauseIcon, playIcon, previousTrackIcon, repeatIcon, shuffle, shuffleIcon, ...rest}) => {
+	render: ({badge, disabled, menuIcon, nextTrackIcon, onNext, onPause, onPlay, onPrevious, onRepeat, onShuffle, paused, pauseIcon, playIcon, previousTrackIcon, repeatIcon, shuffle, shuffleIcon, ...rest}) => {
```
Comment (jeonghee27) [inline MediaPlayer/MediaControls.js:213]: What I don't understand is this is not the sandstone props. The structure is quite different from Sandstone.
Why are you just changing this?
Why Don't you add `spotlight-disabled`?
The title of the PR and the actual edit do not match. so I want know the reason

File: MediaPlayer/MediaPlayer.js [hunk 7]
```diff
@@ -239,6 +264,14 @@ const MediaPlayerBase = kind({
 		 */
 		sourceIndex: PropTypes.number,
 
+		/**
+		 * Disables 5-way spotlight from navigating into the component.
```
Comment (jeonghee27) [inline MediaPlayer/MediaPlayer.js:268]:
```suggestion
		 * Disables 5-way spotlight navigation from navigating into the component.
```

File: tests/screenshot/apps/components/MediaPlayer.js [hunk 8]
```diff
@@ -4,6 +4,9 @@ const MediaPlayerTests = [
 	<MediaPlayer>
 		<source src="https://sampleswap.org/mp3/artist/254731/BossPlayer_Your-Right-Here-160.mp3" type="audio/mp3" />
 	</MediaPlayer>,
+	<MediaPlayer spotlightDisabled>
```
Comment (jeonghee27) [inline tests/screenshot/apps/components/MediaPlayer.js:7]:
```suggestion
	<MediaPlayer disabled>
```

The spotlightDisabled property doesn't change any looks, so I don't think we can verify via screenshot test. The ui test is more appropriate.

File: tests/ui/apps/MediaPlayer/MediaPlayer-View.js [hunk 9]
```diff
@@ -17,14 +17,22 @@ const audioFiles = [
...
 		<Heading>Media Player Default</Heading>
 		<MediaPlayer id="mediaPlayerDefault">
 			{
 				audioFiles.map((audioFile, index) => (<source key={index} src={audioFile} type="audio/mp3" />))
 			}
 		</MediaPlayer>
 	</div>
+	<div style={{marginTop: '80px'}}>
+		<Heading>Media Player SpotlightDisabled</Heading>
+		<MediaPlayer id="mediaPlayerSpotlightDisabled" spotlightDisabled>
```
Comment (jeonghee27) [inline tests/ui/apps/MediaPlayer/MediaPlayer-View.js:30]: Could you please add MediaPlayer with `disabled` case also ??

File: MediaPlayer/MediaControls.js (hunk 5 above)
Comment (jeonghee27) [inline MediaPlayer/MediaControls.js:35]: I know this is from Sandstone, but I want it to be similar to other agate descriptions.

		 * Disables MediaControls and makes it non-interactive.

Or please write it in your way using "disables".

File: CHANGELOG.md (hunk 4 above)
Comment (stanca-pop-lgp) [reply CHANGELOG.md]: It was by mistake.

File: MediaPlayer/MediaControls.js (hunk 6 above)
Comment (stanca-pop-lgp) [reply MediaPlayer/MediaControls.js:213]: I discussed this with the team. In short, we did not choose the proper title for this task/PR. It just focuses on adding the disabled and spotlightDisabled props for the MediaPlayer.
Sandstone does not have a stand-alone component for the MediaPlayer, only for VideoPlayer, so we just used the parts that corresponded to the Agate MediaPlayer design. 
If you guys consider that we should add extra functionality/props, just let us know.

**Review by jeonghee27 — APPROVED** [review]
LGTM

File: MediaPlayer/MediaPlayer.js [hunk 10]
```diff
@@ -261,37 +294,41 @@ const MediaPlayerBase = kind({
...
-					onPause={onPause}
-					onPlay={onPlay}
-					onPrevious={onPrevious}
-					onRepeat={onRepeat}
-					onShuffle={onShuffle}
-					paused={paused}
-					repeat={repeat}
-					shuffle={shuffle}
-				/>
+				<Container spotlightDisabled={spotlightDisabled}>
```
Comment (MikyungKim) [inline MediaPlayer/MediaPlayer.js:300]: Can we not add another div here? I guess we could do like,
<Container {...rest} spotlightDisabled={spotlightDisabled}>

File: tests/ui/specs/MediaPlayer/MediaPlayer-specs.js [hunk 11]
```diff
@@ -6,92 +6,223 @@ describe('MediaPlayer', function () {
...
+			expect(mediaPlayerDisabled.knob.getCSSProperty('left').value).to.equal('0px');
+		});
+
+		it('should not play next media on nextButton click', function () {
+			expect(mediaPlayerDisabled.source).to.equal('https://sampleswap.org/mp3/artist/254731/BossPlayer_Your-Right-Here-160.mp3');
+
+			mediaPlayerDisabled.nextButton.click();
+			Page.delay(1000);
+
+			expect(mediaPlayerDisabled.source).to.not.equal('https://sampleswap.org/mp3/artist/78152/HiatusManJBanner_Show-Stopper-160.mp3');
```
Comment (MikyungKim) [inline tests/ui/specs/MediaPlayer/MediaPlayer-specs.js:125]: Should this be equal to 'https://sampleswap.org/mp3/artist/254731/BossPlayer_Your-Right-Here-160.mp3' for clarity?

**Review by MikyungKim — APPROVED** [review]
LGTM
